### PR TITLE
Fix ARM template template-with-new-rg.json for Slack

### PR DESCRIPTION
**Pull request by BruceHaley**
Fills part of issue #4356

### CODE REVIEW

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Slack.TestBot/DeploymentTemplates/template-with-new-rg.json [hunk 1]
```diff
@@ -117,6 +118,9 @@
         "[resourceId('Microsoft.Resources/resourceGroups/', parameters('groupName'))]"
       ],
       "properties": {
+        "expressionEvaluationOptions": {
+          "scope": "outer"
+        },
```
Comment (mrivera-ms) [inline tests/Adapters/Microsoft.Bot.Builder.Adapters.Slack.TestBot/DeploymentTemplates/template-with-new-rg.json:123]: This is the default behavior. Do not need to add it.

File: build/yaml/botbuilder-dotnet-ci-slack-test.yml [hunk 2]
```diff
@@ -67,13 +67,26 @@ steps:
...
-  displayName: 'Create resources, deploy bot'
+  displayName: 'Create resources'
+  inputs:
+    azureSubscription: $(AzureSubscription)
+    scriptLocation: inlineScript
+    inlineScript: |
+     :: Comment out either Option 1 or Option 2 below. Ultimately, both templates should be exercised to ensure they continue to work.
+     
+     :: Option 1: Use the "new-rg" template:
+     call az deployment sub create --name "$(BotGroup)" --template-file "$(System.DefaultWorkingDirectory)\tests\Adapters\Microsoft.Bot.Builder.Adapters.Slack.TestBot\DeploymentTemplates\template-with-new-rg.json" --location "westus" --parameters appId=$(AppId) appSecret="$(AppSecret)" botId="$(BotName)" botSku=F0 newAppServicePlanName="$(BotName)" newWebAppName="$(BotName)" groupName="$(BotGroup)" groupLocation="westus" newAppServicePlanLocation="westus" slackVerificationToken="$(SlackVerificationToken)" slackBotToken="$(SlackBotToken)"  slackClientSigningSecret="$(SlackClientSigningSecret)"
```
Comment (mrivera-ms) [inline build/yaml/botbuilder-dotnet-ci-slack-test.yml:78]: This option will fail with a template validation error if the template is set to "outer scope".

File: tests/Adapters/Microsoft.Bot.Builder.Adapters.Slack.TestBot/DeploymentTemplates/template-with-new-rg.json (hunk 1 above)
Comment (BruceHaley) [reply tests/Adapters/Microsoft.Bot.Builder.Adapters.Slack.TestBot/DeploymentTemplates/template-with-new-rg.json:123]: True. I left this in so I could remember where this scope setting goes. But I will take it out.

File: build/yaml/botbuilder-dotnet-ci-slack-test.yml (hunk 2 above)
Comment (BruceHaley) [reply build/yaml/botbuilder-dotnet-ci-slack-test.yml:78]: This template with "outer scope" is succeeding. No error. It just required that the "dependsOn" statements be properly formed.

**Pull request merged**